test/functional/session_controller_test.rb

```diff
@@ -0,0 +1,118 @@
+require "test_helper"
+require "gds_api/test_helpers/account_api"
+
+class SessionsControllerTest < ActionController::TestCase
+  include GdsApi::TestHelpers::AccountApi
+  include SessionHelpers
+
+  context "GET sign-in" do
+    setup do
+      stub_account_api_get_sign_in_url(redirect_path: "/bank-holiday")
+    end
+
+    should "redirect the user to the GOV.UK Account service domain" do
+      get :create, params: { redirect_path: "/bank-holiday", state_id: "abc123" }
+      assert_response :redirect
+      assert_equal @response.headers["Location"], "http://auth/provider"
+    end
+
+    should "preserve the _ga tracking parameter if provided" do
+      get :create, params: { redirect_path: "/bank-holiday", state_id: "abc123", _ga: "abc123" }
+      assert_equal @response.headers["Location"], "http://auth/provider?_ga=abc123"
+    end
```

Comment:
Nitpick: if you use different values for the `state_id` and the `_ga`, it'll be clearer that it's really the `_ga` which gets preserved (as this current test would pass if the controller did `"?_ga=#{params[:state_id]}"`_

Reply:
that's a great shout, removed.